### PR TITLE
[TR] Feature/ci optimizations

### PR DESCRIPTION
**Pull request by jmleroux**

### CODE REVIEW

File: features/Context/AssertionContext.php [hunk 1]
```diff
@@ -162,11 +163,14 @@ public function theFieldsShouldBeDisabled($fields)
...
      * @Then /^I should see (?:a )?flash message "([^"]*)"$/
+     *
+     * @throws ExpectationException
      */
     public function iShouldSeeFlashMessage($text)
     {
-        // TODO Flash essages tests temporarily disabled because unstable on CI
+        // TODO Flash messages tests temporarily disabled because unstable on CI
         return true;
+//        $this->getMainContext()->wait(10000, '$(".flash-messages-holder").length > 0');
```
Comment (solivier) [inline features/Context/AssertionContext.php]: is this normal ?

File: features/Context/Page/Base/Grid.php [hunk 2]
```diff
@@ -647,15 +666,18 @@ protected function clickOnFilterToManage($filterName)
...
     protected function clickFiltersList()
     {
-        $filterList = $this
-            ->getElement('Filters')
-            ->find('css', 'a#add-filter-button');
-
-        if (!$filterList) {
+        try {
+            $this->spin(function () {
+                $filterList = $filterList = $this
```
Comment (phaseinducer) [inline features/Context/Page/Base/Grid.php]: typo I think

File: features/Context/Page/Base/Grid.php [hunk 3]
```diff
@@ -669,22 +691,26 @@ protected function clickFiltersList()
...
     {
-        $row = $this->getRow($value);
-        $checkbox = $row->find('css', 'input[type="checkbox"]');
-
-        if (!$checkbox) {
+        try {
+            /** @var NodeElement $checkbox */
+            $checkbox = $this->spin(function () use ($value) {
+                $row = $this->getRow($value);
+                $checkbox = $row->find('css', 'input[type="checkbox"]');
```
Comment (phaseinducer) [inline features/Context/Page/Base/Grid.php]: We need to check that `$row` and `$checkbox` are found here.

File: features/Context/AssertionContext.php (hunk 1 above)
Comment (jmleroux) [inline features/Context/AssertionContext.php]: Yes, our CI platform is not stable enough for now.

File: features/Context/WebUser.php [hunk 4]
```diff
@@ -214,6 +220,8 @@ public function iShouldSeeTheCategoryUnderTheCategory($not, $child, $parent)
      */
     public function iVisitTheTab($tab)
     {
+        $tabLocator = sprintf('$("a:contains(\'%s\')").length > 0;', $tab);
+        $this->wait(30000, $tabLocator);
```
Comment (solivier) [inline features/Context/WebUser.php]: is the wait normal here ?

Comment (jmleroux) [inline features/Context/WebUser.php]: Could be replaced by a 'spin' call, but it's working for now

**Comment by solivier** [comment]
GTM

**Comment by phaseinducer** [comment]
GTM

**Pull request merged**